### ISSUE TITLE
Build error: resource exhausted

### ISSUE THREAD

**Issue by ejsmits** [AmpersandTarski/Ampersand#475]
I am trying to update to 3.5.2, but get this error:
ampersand-3.5.2: build
"fatal: Needed a single revision\n"
"fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.\nUse '--' to separate paths from revisions, like this:\n'git <command> [<revision>...] -- [<file>...]'\n"

WARNING: Execution of 'git' command failed.
BuildInfo_Generated.hs will not contain revision information, and therefore
neither will fatal error messages.
Please check your installation

Warning: Cannot read previously generated src/Database/Design/Ampersand/Prototype/StaticFiles_Generated.hs:
src/Database/Design/Ampersand/Prototype/StaticFiles_Generated.hs: openFile: does not exist (No such file or directory)
This warning should disappear the next time you build Ampersand. If the error persists, please report this as a bug.

setup:
static/zwolle/./extensions/ExcelImport/lib/Classes/PHPExcel/Shared/trend/trendClass.php:
resource exhausted

--  While building package ampersand-3.5.2 using:
      /Users/vananreijk/Ampersand/.stack-work/dist/x86_64-osx/Cabal-1.22.5.0/setup/setup --builddir=.stack-work/dist/x86_64-osx/Cabal-1.22.5.0 build exe:ampersand --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1

---
Any ideas?

**Comment by hanjoosten** [comment]
This looks similar to issue #242.  I assume you are on OSX.? If so, you should install from source files. That is described [here](https://ampersandtarski.gitbooks.io/documentation/content/installation/installing_ampersand.html). Maybe @sjcjoosten can help you if you have problems, because he is the only one I know who compiles ampersand with OSX.

**Comment by ejsmits** [comment]
I am indeed on OSX. Same issue as #242, although i am now using stack to pull the file together. If @sjcjoosten could provide me with a binary file again, that would we great.


Op 1 jul. 2016, om 16:56 heeft Han Joosten <notifications@github.com<mailto:notifications@github.com>> het volgende geschreven:


This looks similar to issue #242<https://github.com/AmpersandTarski/Ampersand/issues/242>. I assume you are on OSX.? If so, you should install from source files. That is described here<https://ampersandtarski.gitbooks.io/documentation/content/installation/installing_ampersand.html>. Maybe @sjcjoosten<https://github.com/sjcjoosten> can help you if you have problems, because he is the only one I know who compiles ampersand with OSX.

**Comment by hanjoosten** [comment]
@sjcjoosten begrijp jij wat hier precies misgaat?

**Comment by sjcjoosten** [comment]
Only a limited number of files can be open simultaneously. On Mac OS X, this number is rather low compared to other systems.

Fix for other people running into this problem: use this command in your terminal before installing ampersand:
```
sudo ulimit -n 2048
```
(let me know if that helped, I'm not running into this problem anymore myself)

Fix for ampersand: make sure that we do not leave files open, that is: we should close files after opening them. I would be surprised if there was a reason to have more than 2 files open at a time. Note that in this case it is the installer which is opening multiple files at once.

**Comment by ejsmits** [comment]
:( It didn't work out... I still get the same error. Any other ideas?

**Comment by hanjoosten** [comment]
Could it be that there is nothing wrong with Ampersand, but that your Git doesn't work properly?
You wrote:
~~~~
I am trying to update to 3.5.2, but get this error:
ampersand-3.5.2: build
"fatal: Needed a single revision\n"
"fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.\nUse '--' to separate paths from revisions, like this:\n'git [...] -- [...]'\n"
~~~~

This suggest that you have problems getting the Ampersand source code in place, even before building. What exactly are you doing to install Ampersand?

**Comment by ejsmits** [comment]
Thanks! I got rid of the “fatal” error…. but I am left over with the error below it:

Warning: Cannot read previously generated src/Ampersand/Prototype/StaticFiles_Generated.hs:
src/Ampersand/Prototype/StaticFiles_Generated.hs: openFile: does not exist (No such file or directory)
This warning should disappear the next time you build Ampersand. If the error persists, please report this as a bug.

setup:
static/zwolle/./extensions/ExcelImport/lib/Classes/PHPExcel/Shared/Escher:
resource exhausted

--  While building package ampersand-3.6.1 using:
      /Users/vananreijk/Ampersand/Ampersand/.stack-work/dist/x86_64-osx/Cabal-1.22.5.0/setup/setup --builddir=.stack-work/dist/x86_64-osx/Cabal-1.22.5.0 build lib:ampersand exe:ampersand --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1

I even tried to install it on a different machine (installed brew, installed stack, etc.). But that did not help either: i get the same error. I really would love to start working again with Ampersand… But this does not work out yet.

**Comment by hanjoosten** [comment]
The warning is ok. That's because ampersand is built for the first time. 

I get confused what you are doing. Are you using Stack? You should! What exactly is the command(s) you give? The log you provide doesn't show, as far as I can tell.

**Comment by ejsmits** [comment]
I am giving the command “stack install”. It al works fine, until the end when this error appears.

**Comment by ejsmits** [comment]
Wow, it works now! I think that I messed up my installation of stack (Since I wanted to install the new version of Ampersand into another directory that the one i had the previous version working in?) But I am not sure. Anyway, I removed all previous files (and made a backup of my .adl files), checked my PATH, ran stack install again and installed composer again.
Turned out to be something in my configuration, not in the ampersand-files!
